Library/Homebrew/utils/gems.rb

```diff
@@ -51,8 +51,13 @@ def odie_if_defined(message)
   end
 
   def setup_gem_environment!(gem_home: nil, gem_bindir: nil, setup_path: true)
+    library_path = if ENV["HOMEBREW_STACKPROF"]
+      "#{ENV["HOMEBREW_LIBRARY"]}/Homebrew"
```

Comment:
Is it possible for `HOMEBREW_LIBRARY` to not be set, here? If so, it may be better to determine this path using `__dir__`'s parent